Proposal: Add ephemeral flag to wakuMessage (#532)
vacp2p/rfc: content/docs/rfcs/14/README.md

This field holds the Unix epoch time in nanoseconds. 
Omitting it means the timestamp is unspecified.

> The `ephemeral` field MAY be filled out to signify the transient nature of the message.

Reviewer: ```suggestion
The `ephemeral` field MAY be set to `true` to signify the transient nature of the message.
```

Author: Addressed in [eeed3e2](https://github.com/vacp2p/rfc/pull/532/commits/eeed3e28fef460452ea7a4c58084d6ad7cf3f2e7)

> Omitting it means the message is permananent, and must be added to the store.

Reviewer: I wouldn't say `permanent`. It just indicates that the message COULD be stored by store nodes. If this message SHOULD be stored, this field MUST be set to `false`. (Since `false` is the default for `bool` fields in proto3, omitting it will have the same effect.)

Author: Nice catch! Amending!

Author: Addressed in [eeed3e2](https://github.com/vacp2p/rfc/pull/532/commits/eeed3e28fef460452ea7a4c58084d6ad7cf3f2e7)

> Omitting it means the message is permananent, and must be added to the store.

Reviewer 2: I'd also put a link to the store protocol where it says:
> and must be added to the store.

Also, it makes it clearer to say `store protocol` instead of `store` as one may have no context about the store protocol when reading the WakuMessage specs.

Author: Addressed in [f7983f6](https://github.com/vacp2p/rfc/pull/532/commits/f7983f6f64459863e925478cad8d030bab50bf26)

and [69a8f7c](https://github.com/vacp2p/rfc/pull/532/commits/69a8f7cc719e2329b31ec320c69a234dbfe78c1c)

> Omitting it means the message is permananent, and must be added to the store.

Reviewer 2: If the `false` value is to be explained here, then would be good to also explain what is the interpretation of the `true` value in the context of the store protocol. Borrowing the description from above:
> The message MAY be ignored by the store service node"

Reviewer 2: Alternatively, the reader can be referred to the store protocol specs for how this flag is rendered.

Author: Addressed in [eeed3e2](https://github.com/vacp2p/rfc/pull/532/commits/eeed3e28fef460452ea7a4c58084d6ad7cf3f2e7)

This field holds the Unix epoch time in nanoseconds. 
Omitting it means the timestamp is unspecified.

The `ephemeral` field MAY be set to signify the transient nature of the message.

> If the message SHOULD be stored by the [store protocol](/spec/13), then this field MUST be set to `false`, which is equivalent to omitting the field.

Reviewer 3: ```suggestion
If the message should be stored by the [store protocol](/spec/13), then this field MUST be set to `false`, which is equivalent to omitting the field.
```

Reviewer 3: This is not a SHOULD specified by the RFC but a should decided by the client / user.

Author: Addressed in [2d56fe8](https://github.com/vacp2p/rfc/pull/532/commits/2d56fe8d8778302f0c0f7886e18dfbe9de745020)

> If the message SHOULD NOT be stored by the [store protocol](/spec/13), then this field MUST be set to `true`.

Reviewer 3: ```suggestion
If the message should not be stored by the [store protocol](/spec/13), then this field MUST be set to `true`.
```

Reviewer 3: see above

Author: Addressed in [49d8758](https://github.com/vacp2p/rfc/pull/532/commits/49d87588ca1249b48a7769e0afb5d9ec5ae497fd)
